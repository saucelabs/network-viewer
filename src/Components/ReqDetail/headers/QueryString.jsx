import React from 'react';
import PropTypes from 'prop-types';

import Styles from './../Headers.styles.scss';

const QueryString = ({ data, isPayloadTransformed }) => (
  <div className={Styles['header-detail']}>
    {data.headers.queryString.map(({ name, value }, index) => (
      <p
        key={`${name}-${index}`}
        className={Styles['info-row']}
      >
        <span className={Styles['info-caption']}>
          {`${name}:`}
        </span>
        <span className={Styles['info-value']}>
          {isPayloadTransformed ? decodeURIComponent(value) : value}
        </span>
      </p>
    ))}
  </div>
);

QueryString.propTypes = {
  data: PropTypes.object,
  isPayloadTransformed: PropTypes.bool.isRequired,
};

QueryString.defaultProps = {
  data: null,
};

export default QueryString;
