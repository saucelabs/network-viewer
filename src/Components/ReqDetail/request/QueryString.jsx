import React from 'react';
import PropTypes from 'prop-types';

import Styles from '../ReqDetail.styles.scss';

const QueryString = ({ data, isPayloadTransformed }) => (
  <div className={Styles['section-detail']}>
    {data.headers.queryString.map(({ name, value }, index) => (
      <div
        key={`${name}-${index}`}
        className={Styles['info-row']}
      >
        <span className={Styles['info-caption']}>
          {`${name}:`}
        </span>
        <span className={Styles['info-value']}>
          {isPayloadTransformed ? decodeURIComponent(value) : value}
        </span>
      </div>
    ))}
  </div>
);

QueryString.propTypes = {
  data: PropTypes.object,
  isPayloadTransformed: PropTypes.bool.isRequired,
};

QueryString.defaultProps = {
  data: null,
};

export default QueryString;
