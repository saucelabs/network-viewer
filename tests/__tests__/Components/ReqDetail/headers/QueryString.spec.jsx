import React from 'react';
import { shallow } from 'enzyme';

import QueryString from '../../../../../src/Components/ReqDetail/sections/QueryString';

describe('QueryString', () => {
  const props = {
    data: {
      headers: {
        queryString: [{
          name: 'foo',
          value: 'bar',
        }],
      },
    },
    isPayloadTransformed: true,
  };

  it('renders without crashing', () => {
    const element = shallow(<QueryString {...props} />);
    expect(element).toMatchSnapshot();
  });
});
