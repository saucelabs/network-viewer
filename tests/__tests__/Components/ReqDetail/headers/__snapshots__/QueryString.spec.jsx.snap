// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryString renders without crashing 1`] = `
<div>
  <div
    key="foo-0"
  >
    <span>
      foo:
    </span>
    <span>
      bar
    </span>
  </div>
</div>
`;
